src/lib/comms/sol-socket-dtls-impl-tinydtls.c

```diff
@@ -160,16 +146,21 @@ session_from_linkaddr(const struct sol_network_link_addr *addr,
 }
 
 static void
+clear_queue_item(struct queue_item *item)
+{
+    sol_util_secure_clear_memory(item->buffer.data, item->buffer.capacity);
```

Comment:
This has been done before sol_buffer had a flag to clear memory securely. Please set this flag in the buffer so this is done automatically. Note that this only works when sol_buffer does not own the memory due to realloc() possibly keeping the old data around when the pointer changes; you might need to fix that first (basically, if that flag is set, instead of calling realloc(), always allocate a new block, clear the previous one -- using the previous capacity information from the buffer struct -- and free the previous block).